aptos-move/dynamic-transaction-composer/src/codegen.rs

```diff
@@ -0,0 +1,380 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    ArgumentOperation, BatchArgument, BatchedFunctionCall, PreviousResult, APTOS_SCRIPT_BUILDER_KEY,
+};
+use move_binary_format::{
+    access::{ModuleAccess, ScriptAccess},
+    builders::CompiledScriptBuilder,
+    errors::{PartialVMError, PartialVMResult},
+    file_format::*,
+};
+use move_core_types::{
+    identifier::IdentStr,
+    language_storage::{ModuleId, TypeTag},
+    metadata::Metadata,
+    transaction_argument::TransactionArgument,
+    vm_status::StatusCode,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Default)]
+struct Context {
+    script: CompiledScript,
+    // Arguments to the compiled script.
+    args: Vec<Vec<u8>>,
+    // (nth move call) -> (type arguments for the call)
```

Comment:
Instead of N arrays of types T1..TN, why not have one array of type struct{T1..TN}. This seems to be much cleaner.